2019-02-06  Brendon Tetsuo Ng  <classvwt@lnxsrv07.seas.ucla.edu>

        * lib/timespec.h (timespec_cmp): Use comparison operators to prevent overflow
        due to subtraction that occurred when dealing with files very long apart

--- lib/timespec.h	2019-02-06 00:11:36.684731000 -0800
+++ lib/timespec.h	2019-02-06 00:23:00.649676000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec == b.tv_sec)
+    return 0;
+  else
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
